### ISSUE TITLE
Watch for changes - stops reacting if file changes multiple times

### ISSUE THREAD

**Issue by Juraj-Masiar** [mozilla/web-ext#2249]
### Is this a feature request or a bug?
Bug

### What is the current behavior?
1) using any project
1) open console and navigate to the folder where web-ext is running
1) execute `touch some_file` multiple times in a short succession (every 500ms)
```touch 128.png```
1) notice how the addon got reloaded only once?

### What is the expected or desired behavior?
It would be nice if the addon got reload after the last change.
The reload throttle is fine, but if there are no more changes being reported, it should reload it again. Right?

### Version information (for bug reports)

* **Firefox version**: 89
* **Your OS and version**: Win 21H1
* Paste the output of these commands:

````
v14.16.1
6.14.11
6.1.0
````

https://user-images.githubusercontent.com/6358971/120074132-b5f30a80-c09b-11eb-9ffb-bf7c29f7882f.mp4

The reason I'm reporting this is actually different. Sometimes when I make a change, the web-ext won't reload the addon at all. It's like 50% chance it will or won't reload. As I was experimenting with it, I've noticed this issue.
After A LOT of work I've managed to hotfix it by using the `--watch-file version` and by touching a dummy `version` file when the webpack makes changes.

**Comment by Rob--W** [comment]
To avoid too many repeated reloads, the reload is debounced using [debounce](https://www.npmjs.com/package/debounce), i.e. additional changes within (by default) 1 second are ignored: https://github.com/mozilla/web-ext/blob/8bd0d13f425bbaed6cfc2f8127883c5c43010f1d/src/watcher.js#L61

What you're asking for is to refresh immediately for the first change, and if anything has happened in the following second, to reload once more thereafter. When the source changes files many times, this would then always result in two reloads. This is not ideal either.

To have predictable reloading behavior when a build tool is used, touching a file after the completion of a build is recommended (in combination with `--watch-file`).

**Comment by Juraj-Masiar** [comment]
Actually what I would prefer is to defer the execution by some time (~400ms) and postpone it by this time every time it's called again within the delay (so that it's called once after this time when things stops changing).

**The current behavior is not stable for me, is it for you?** Sometimes the addon reloads broken or disappears completely and then reappears after further changes. 
I use Webpack 4, maybe I need to upgrade to 5?

Anyway, the workaround with watching a file seems to be working fine so it's OK for me. I use [webpack-touch](https://www.npmjs.com/package/webpack-touch).

**Comment by Rob--W** [comment]
> Actually what I would prefer is to defer the execution by some time (~400ms) and postpone it by this time every time it's called again within the delay (so that it's called once after this time when things stops changing).

Sounds reasonable to me.

> **The current behavior is not stable for me, is it for you?** Sometimes the addon reloads broken or disappears completely and then reappears after further changes.

That could happen if the reload is initiated while the add-on is being rebuild.
